airflow/models/dag.py

```diff
@@ -304,14 +339,15 @@ def __init__(
 
         # set timezone from start_date
         if start_date and start_date.tzinfo:
-            self.timezone = start_date.tzinfo
+            self._schedule_timezone = _get_tzname_or_offset(start_date.tzinfo)
         elif 'start_date' in self.default_args and self.default_args['start_date']:
             if isinstance(self.default_args['start_date'], str):
                 self.default_args['start_date'] = timezone.parse(self.default_args['start_date'])
-            self.timezone = self.default_args['start_date'].tzinfo
+            if self.default_args['start_date'] and self.default_args['start_date'].tzinfo:
+                self._schedule_timezone = _get_tzname_or_offset(self.default_args['start_date'].tzinfo)
 
-        if not hasattr(self, 'timezone') or not self.timezone:
-            self.timezone = settings.TIMEZONE
+        if not hasattr(self, '_schedule_timezone'):
+            self._schedule_timezone = _get_tzname_or_offset(settings.TIMEZONE)
```

Comment:
Does any of this actualy need changing?

Can't we keep `self.timezone` as an actual timezone object?

Reply:
sure